### ISSUE TITLE
Fatal Error & Segfault (Access to undeclared static property)

### ISSUE THREAD

**Issue by Nuranto** [nexcess/magento-turpentine#1070]
I've updated turpentine to last version (was the last when did it : 0.6.6) on one of my client magento.
Since then, I've got segfaults, and site crash. It is chaotic : everything works fine for days, and one day, crash.

The error I'm able to see is : 
Access to undeclared static property: Varien_Object::$_underscoreCache
I've also seen this one : 
PHP Fatal error:  Access to undeclared static property: Mage::$_isDeveloperMode (note : in Mage.php)
And this one : 
tal error: Call to a member function getCode() on a non-object in app/code/core/Mage/Customer/Model/Session.php on line 71 (note : non-object = Mage::app()->getStore()->getWebsite() oO)

I was not sure it was coming from turpentine, and actually i'm still not sure. But I freshly installed turpentine (0.6.7) on another website and I've got the same errors since. So.. 

Is anybody else have those problems ?

EDIT : But... I have 0.6.6 on a third magento, and i do not have this kind of error.
First magento (with error) : Turpentine 0.6.7, Apache 2.2.22-13+deb7u6, PHP 5.4.40-1~dotdeb+wheezy.1
Second magento (with error) : Turpentine 0.6.6, Apache 2.2.22, PHP 5.4.45-0+deb7u2
Third magento (without error) : Turpentine 0.6.6, Apache 2.4.10, PHP 5.5.12-2ubuntu4.2

=> Is it a PHP 5.4 related error ?

**Comment by miguelbalparda** [comment]
There are a lot of differences between your environments. It seems you are the only one having this issue so far, so I would tend to say this might be an environment issue instead of a Turpentine issue at least for now. 
I'll try to replicate your issue, but I suspect this is something a lot of users will face if the issue was in the code.

**Comment by Nuranto** [comment]
Alright, as I said, I'm not sure it is turpentine-related.
But well, if someone else have this kind of issues with turpentine, please answer to this post.
I will continue my search on my side!

**Comment by ADDISON74** [comment]
A friend of mine wanted to re-install Magento from scratch but keeping the existent files. I deleted all the tables into the database, deleted /app/etc/local.xml file and folders /var/cache, var/session and var/report/. I got 502 server error because of this issue in Apache log: "Call to a member function getCode() on a non-object in app/code/core/Mage/Customer/Model/Session.php on line 71". Initially it was a strange situation but I did this many times and it was working. So I started checking if there are any extensions. I found Turpentine. I set active to false and I could reinstall Magento from scratch without any issue. 

My advice is removing completely any trace of Turpentine from that server and see if it is working without Turpentine. In a VM test your configuration with devel version. I am using it in production even it is called development. I am using Pound in front of Varnish for SSL offloading. If something bad is happening with Varnish caching I can skip it in a minute from Pound configuration.
